src/views/Notifications/Home/_components/_utils.tsx

```diff
@@ -0,0 +1,194 @@
+        <Icon.ExpandMore className="margin-left-1" aria-hidden />
+      </>
+    );
+  }
+  return <></>;
+};
+
+type ChangesArrayType =
+  | 'oldName'
+  | 'added'
+  | 'count'
+  | 'crTdls'
+  | 'planDiscussions'
+  | 'readyForReview'
+  | 'readyForClearance'
+  | 'updated'
+  | 'statusChanges';
+
+/**
+ * In the Daily Digest Notification center, we want to show only the first five changes and hide everything else under a `+X more changes`.
+ *
+ * In order to determine what actually changed, this function does the following:
+ * * combs through the `changes` object
+ * * filters out any values that are empty string, null, and empty arrays, indicating there is no change
+ * * puts remaining values into an array.
+ *
+ * Further splicing of the array will then only show the first five changes and the rest will be under the `+X more changes`
+ *
+ * A few exceptions:
+ * * Model Lead changes are important and prepending it to the front of the array ensures that a model lead addition will always show up, rather than being spliced out.
+ *     * Model Lead being added to the Model Plan. The object looks like the following:
+ *     ```
+ *        "modelLeads": {
+            "added": [
+                {
+                    ...
+                    "commonName": "MINT Doe"
+                }
+            ]
+          },
+        ```
+ * * `crTdls` and `planDisucssions`'s object are identical. So rather than having its object go through the filter, this function will push the key if it is 'crTdls' & 'planDiscussions'
+
+      ```
+      "crTdls": {
+          "activity": true
+      },
+      "planDiscussions": {
+          "activity": true
+      }
+      ```
+ *
+ *
+ * @param {object} changes object passed in from `analyzedAudits`.
+ * @returns `ChangesArrayType[]`
+ */
+export const pushValuesToChangesArray = (
+  obj: ChangeTypes
+): ChangesArrayType[] => {
+  const changesArray: ChangesArrayType[] = [];
+
+  const pushValues = (innerObj: ChangeTypes) => {
+    Object.entries(innerObj).forEach(([key, value]) => {
+      if (key !== '__typename') {
+        // Ignore key values that where the key is '__typename'
+        if (Array.isArray(value) && value.length > 0) {
+          // Ignore empty arrays (i.e. [])
+          if (key === 'added') {
+            // Model Lead added is important, so pushed to the front of array to ensure it will always be shown
+            changesArray.unshift(key);
+          } else {
+            changesArray.push(key as ChangesArrayType);
+          }
+        } else if (typeof value === 'string' && value.trim() !== '') {
+          // Push the key if only the value is not an empty string (i.e. '')
+          changesArray.push(key as ChangesArrayType);
+        } else if (typeof value === 'number') {
+          changesArray.push(key as ChangesArrayType);
+        } else if (typeof value === 'boolean' && value) {
+          changesArray.push(key as ChangesArrayType);
```

Comment:
@patrickseguraoddball not sure if this is the best way, but when I made the types for the ChangesArray more strict, `key` was throwing the follow type error:

```
Argument of type 'string' is not assignable to parameter of type 'ChangesArrayType'.ts(2345)
```

Reply:
ICYWW

```
type ChangesArrayType =
  | 'oldName'
  | 'added'
  | 'count'
  | 'crTdls'
  | 'planDiscussions'
  | 'readyForReview'
  | 'readyForClearance'
  | 'updated'
  | 'statusChanges';
```